cmd/root.go

```diff
@@ -107,10 +107,6 @@ func NewRootCmd() *cobra.Command {
 		longDesc = longDesc + "\n" + build.EulaNotice
 	}
 	cmd.Long = longDesc
-
-	// workaround for the data-dir location input for the kubectl command
-	cmd.PersistentFlags().AddFlagSet(config.GetKubeCtlFlagSet())
```

Comment:
Don't quite get how this solves the original issue, care to explain a bit?

Reply:
The name is a bit misleading (not sure why it was named like that originally) but `GetKubeCtlFlagSet` contains the global flags data-dir and debug (https://github.com/k0sproject/k0s/blob/f639051d29e31b7d7f6b31dcd30171fb882a6051/pkg/config/cli.go#L121-L126).
By removing it from from root.go, we basically ensure that only commands that need these flags include them. When this was in the root.go, these flags were inherited by all subcommands, which we don't want them to.